more clang-tidy

neheb opened the pull request.

neheb committed:
clang-tidy: apply to newly merged stuff

Signed-off-by: Rosen Penev <rosenp@gmail.com>

neheb committed:
use some future C++ stuff when possible

Signed-off-by: Rosen Penev <rosenp@gmail.com>

neheb committed:
static_cast make_unsigned

Signed-off-by: Rosen Penev <rosenp@gmail.com>

neheb commented:
```
FAIL: test_run (test_exifprint_lint.test_exifprint_lint.test_run)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\exiv2\exiv2\tests\system_tests.py", line 652, in test_run
    self.compare_stderr(i, command, processed_stderr, stderr)
  File "D:\a\exiv2\exiv2\tests\system_tests.py", line 773, in compare_stderr
    self._compare_output(
  File "D:\a\exiv2\exiv2\tests\system_tests.py", line 745, in _compare_output
    self.assertMultiLineEqual(
AssertionError: '' != 'Exif.Nikon3.ExposureTuning type Undefined (7) expected Short (3)\n'
+ Exif.Nikon3.ExposureTuning type Undefined (7) expected Short (3)
 : Standard error does not match
```

I'm curious why...

neheb commented:
Got it. Bad simplification of boolean. Manually adjusted.

postscript-dev commented:
@neheb 
I noticed that your recent PR and this one correct a formatting errors by adding an extra space. E.g.
```
- //#ifdef SOMETHING
+ // #ifdef SOMETHING
```
Maybe `.clang-format` could be changed to automatically catch this in future? If appropriate, the change could be made in a different PR if you prefer.

neheb commented:
@postscript-dev I don't really see it elsewhere except for the recently committed video PRs.

Maybe https://github.com/Exiv2/exiv2/pull/2448 should be merged first.

piponazo commented:
> Maybe `.clang-format` could be changed to automatically catch this in future? If appropriate, the change could be made in a different PR if you prefer.

Yes please, if you think this would reduce friction with PRs, please propose a different PR and explain the situation.

piponazo commented on include/exiv2/value.hpp:
```diff
@@ -1270,16 +1270,28 @@ class ValueType : public Value {
     }
 
     // Check for integer overflow.
+#if __cplusplus >= 201703L
```
[question] Does it make sense to add all these conditional code? We are already indicating in the CMake code that we need a C++17 compiler for this project. Therefore, I do not see the point of having conditionals for older versions of the C++ standard.

neheb replied:
This is for the headers. It was requested earlier that they be kept compatible with C++11.

neheb committed:
get rid of -fanalyzer memory leaks

Don't use make_shared inside a function. Instead, change constructor to
value to have std::move.

Also move shared_ptrs everywhere. It's fairly expensive to copy.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

neheb commented:
ran through GCC's -fanalyzer. Most obvious issue is pointless and potentially problematic shared_ptr copying. Fixed.

piponazo reviewed (approved):
LGTM thanks!
